lib/test-worker.js

```diff
@@ -30,6 +30,23 @@ process.installPrecompilerHook();
 const dependencies = [];
 process.installDependencyTracking(dependencies, testPath);
 
+if (opts.extensions) {
+	opts.extensions.forEach(function (ext) {
+		switch (ext) {
+			case 'ts':
+				var extTs = require('./ext/ts');
+				extTs.register();
+				break;
+			case 'js':
+				// We already have babel for ES6/7
+				break;
+
+			default:
+				throw new Error('AVA: `--extension=' + ext + '` is not supported by AVA.');
+		}
+	});
+}
```

Comment:
I don't think the TypeScript support would still work after we removed those changes. Because if no `extTs.register()`, then test-worker.js will not work with .ts test files.

This TypeScript support will only be enabled when we specify `--extension ts` arg with AVA, which is the case we need to automatically transpile all TypeScript file with extension: `.ts`

And about the dependencies, I think for TypeScript is different as for JavaScript, because:
1. There are almost no TypeScript dependencies in NPM(node_modules)
1. The `.ts` extension is not the default JavaScript file extension, so it's almost only in our development environment.

Please correct me if I'm wrong, or missed anything. Thanks.